working/4271 - static enough metaprogramming/proposal.md

```diff
@@ -0,0 +1,1222 @@
+tldr: I propose we follow the lead of **D**, **Zig** and **C++26** when it comes
```

Comment:
I'd give this file a header.

- A title line at the top.
- An author line: `Author: @mralpeh`.
- Consider a version line: `version: 1.0` and a version section/changelog at the end, to track changes.